data/pigui/libs/modal-win.lua

```diff
@@ -7,22 +7,19 @@ local modals = {}
 local modalStack = {}
 
 local ModalWindow = {}
+local defaultModalFlags = ui.WindowFlags {"NoTitleBar", "NoResize", "AlwaysAutoResize", "NoMove"}
 
 function ModalWindow.New(name, innerHandler, outerHandler, flags)
 	local self = {
 		name = name,
-		flags = flags or {"NoTitleBar", "NoResize", "AlwaysAutoResize", "NoMove"},
+		flags = flags or defaultModalFlags,
 		stackIdx = -1,
 		isOpen = false,
 		innerHandler = innerHandler,
-		outerHandler = outerHandler,
-	}
-
-	if not outerHandler then
-		self.outerHandler = function(drawPopupFn)
+		outerHandler = outerHandler or function(slf, drawPopupFn)
```

Comment:
This is the world's tiniest nitpick, but I'd recommend using `self` for consistency.
```suggestion
		outerHandler = outerHandler or function(self, drawPopupFn)
```

Reply:
Fixed and squashed

Reply:
I also renamed the new instance variable to modalWin, it was the whole reason for naming the function argument "slf" - didn't want to confuse people.